protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -190,7 +153,7 @@ open class CodeGenerator(val file: File, val kotlinTypeMappings: Map<String, Str
             } else if (field.type == File.Field.Type.MESSAGE) {
                 line("${field.kotlinFieldName} = " +
                     "${field.kotlinFieldName}?.plus(plus.${field.kotlinFieldName}) ?: plus.${field.kotlinFieldName},")
-            } else if (file.version == 2 && field.optional) {
+            } else if (field.hasPresence) {
```

Comment:
`hasPresence` means that its optional?

Reply:
Yeah. This is the new terminology that protobuf is using now that they added `optional` support in proto3. If you're curious, this has more details: https://github.com/protocolbuffers/protobuf/blob/master/docs/field_presence.md.